src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -175,13 +175,17 @@ internal void CloseAsIs()
                     SpinWait sw = new SpinWait();
                     while (!_released)
                     {
+                        // The socket was not released due to the SafeHandle being used.
+                        // Try to make those on-going calls return.
+                        // On Linux, TryUnblockSocket will unblock current operations but it doesn't prevent
+                        // a new one from starting. So we must call TryUnblockSocket multiple times.
```

Comment:
I'm not understanding.  TryUnblockSocket calls Disconnect.  If an operation starts after Disconnect is called, that operation doesn't immediately fail?

Reply:
On Linux, after Disconnect the socket can be re-used.